alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -245,118 +201,99 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
                             indexLastRunContext[shard] = max(-1, (indexUpdatedRunContext[shard] as String).toInt() - 10)
                         }
                     }
-
-                    val fieldsToBeQueried = mutableSetOf<String>()
-                    if (monitorCtx.fetchOnlyQueryFieldNames) {
-                        for (it in queries) {
-                            if (it.queryFieldNames.isEmpty()) {
-                                fieldsToBeQueried.clear()
-                                logger.debug(
-                                    "Monitor ${monitor.id} : " +
-                                        "Doc Level query ${it.id} : ${it.query} doesn't have queryFieldNames populated. " +
-                                        "Cannot optimize monitor to fetch only query-relevant fields. " +
-                                        "Querying entire doc source."
-                                )
-                                break
-                            }
-                            fieldsToBeQueried.addAll(it.queryFieldNames)
-                        }
-                        if (fieldsToBeQueried.isNotEmpty())
-                            logger.debug(
-                                "Monitor ${monitor.id} Querying only fields " +
-                                    "${fieldsToBeQueried.joinToString()} instead of entire _source of documents"
-                            )
-                    }
                     val indexExecutionContext = IndexExecutionContext(
                         queries,
                         indexLastRunContext,
                         indexUpdatedRunContext,
                         updatedIndexName,
                         concreteIndexName,
+                        updatedIndexNames,
+                        concreteIndices,
                         conflictingFields.toList(),
                         matchingDocIdsPerIndex?.get(concreteIndexName),
                     )
 
-                    fetchShardDataAndMaybeExecutePercolateQueries(
-                        monitor,
+                    val shards = mutableSetOf<String>()
```

Comment:
**parallelization should be done for entire set of indices not per index.**

Reply:
the no. of requests made doesnt significantly differ in this approach compared to doing parallelization on entire set of indices.
e.g. if there are `2 indices` having `11 & 7` shards respectively & if cluster has `5 nodes`, if we're doing parallelization on individual indices, 
`11 / 5 + 7 / 5` i.e. `5 calls` instead of `(11 + 7) / 5` i.e. `4 calls`.

Reply:
the example you took is not at scale.